src/uu/chcpu/src/chcpu.rs

```diff
@@ -0,0 +1,379 @@
+        if configured.trim() == "1" {
+            println!("CPU {} is already configured", self.0);
+            return;
+        };
+
+        let result = File::create(cfg_path).and_then(|mut f| f.write_all(b"1"));
+        match result {
+            Ok(_) => println!("CPU {} configured", self.0),
+            Err(e) => println!("CPU {} configure failed: {}", self.0, e.kind()),
+        };
+    }
+
+    fn deconfigure(&self) {
+        if !self.is_configurable() {
+            println!("CPU {} is not configurable", self.0);
+            return;
+        }
+
+        let cfg_path = self.get_path().join("configure");
+
+        let configured = fs::read_to_string(&cfg_path).unwrap();
+        if configured.trim() == "0" {
+            println!("CPU {} is already deconfigured", self.0);
+            return;
+        };
+
+        if self.is_online() {
+            println!("CPU {} deconfigure failed (CPU is enabled)", self.0);
+            return;
+        }
+
+        let result = File::create(cfg_path).and_then(|mut f| f.write_all(b"0"));
+        match result {
+            Ok(_) => println!("CPU {} deconfigured", self.0),
+            Err(e) => println!("CPU {} deconfigure failed: {}", self.0, e.kind()),
+        };
+    }
+}
+
+fn get_online_cpus() -> Vec<Cpu> {
+    let cpu_list = fs::read_to_string("/sys/devices/system/cpu/online").unwrap();
+    parse_cpu_list(&cpu_list).iter().map(|n| Cpu(*n)).collect()
+}
+
+fn trigger_rescan() {
+    let path = PathBuf::from("/sys/devices/system/cpu/rescan");
+
+    if !path.exists() {
+        // TODO: This should exit gracefully with a ExitCode::FAILURE instead of quietly returning
+        println!("This system does not support rescanning of CPUs");
+        return;
+    }
+
+    let result = File::create(path).and_then(|mut f| f.write_all(b"1"));
+    match result {
+        Ok(_) => println!("Triggered rescan of CPUs"),
+
+        // TODO: This needs to exit with ExitCode::FAILURE
+        Err(e) => println!("Failed to trigger rescan of CPUs: {}", e.kind()),
+    };
+}
+
+fn enable_cpus(cpu_list: &str) {
+    let to_enable = parse_cpu_list(cpu_list).into_iter().map(Cpu);
+
+    for cpu in to_enable {
+        if cpu.ensure_exists() {
+            cpu.enable();
+        };
+    }
+}
+
+fn disable_cpus(cpu_list: &str) {
+    let to_disable = parse_cpu_list(cpu_list).into_iter().map(Cpu);
+
+    for cpu in to_disable {
+        if cpu.ensure_exists() {
+            cpu.disable();
+        }
+    }
+}
+
+fn configure_cpus(cpu_list: &str) {
+    let to_configure = parse_cpu_list(cpu_list).into_iter().map(Cpu);
+    for cpu in to_configure {
+        if cpu.ensure_exists() {
+            cpu.configure();
+        }
+    }
+}
+
+fn deconfigure_cpus(cpu_list: &str) {
+    let to_deconfigure = parse_cpu_list(cpu_list).into_iter().map(Cpu);
+    for cpu in to_deconfigure {
+        if cpu.ensure_exists() {
+            cpu.deconfigure();
+        }
+    }
+}
```

Comment:
what about something like ?

```suggestion
fn process_cpus<F>(cpu_list: &str, action: F)
where
    F: Fn(&Cpu)
{
    parse_cpu_list(cpu_list)
        .into_iter()
        .map(Cpu)
        .filter(|cpu| cpu.ensure_exists())
        .for_each(|cpu| action(&cpu));
}

// Usage becomes:
fn enable_cpus(cpu_list: &str) {
    process_cpus(cpu_list, |cpu| cpu.enable());
}

fn disable_cpus(cpu_list: &str) {
    process_cpus(cpu_list, |cpu| cpu.disable());
}

fn configure_cpus(cpu_list: &str) {
    process_cpus(cpu_list, |cpu| cpu.configure());
}

fn deconfigure_cpus(cpu_list: &str) {
    process_cpus(cpu_list, |cpu| cpu.deconfigure());
}
```